tests/jest-interactor/click-callback/test.js

```diff
@@ -0,0 +1,22 @@
+// highlight{7,13,18-21}
+import React from 'react';
+import { Button } from 'shared/components/Button';
+import Interactor from '@bigtest/interactor';
+import { mount } from '@bigtest/react';
+
+let button = new Interactor('button');
+
+it('calls "onClick" prop on button click', async () => {
+  // Render new instance in every test to prevent leaking state
+  const onClick = jest.fn();
+  // Mount the component in the DOM
+  await mount(() => <Button onClick={onClick} />);
+
+  // Use interactor `when` to assert the change has happened
+  // this will loop until `true` & fails if it isn't true by
+  // the timeout (2s by default)
+  button.when(() => {
+    expect(button.text).toEqual('Click me nao');
+    expect(onClick).toHaveBeenCalled();
```

Comment:
How does this work? I don't see where you trigger the click event.

Reply:
Heh, this was actually a bug on my part. I missed an `await` so it ever got run. Since it never ran, it never failed for not clicking 😱.  The correct code is `await button.click().when...`